syntax.php

```diff
@@ -58,18 +72,27 @@ public function handle($match, $state, $pos, Doku_Handler &$handler) {
      */
     public function render($mode, Doku_Renderer &$renderer, $data) {
         if($mode == 'metadata') return false;
+        if($data[0] != DOKU_LEXER_EXIT) return false;
 
         /** @var helper_plugin_top $hlp */
         $hlp  = plugin_load('helper', 'top');
-        $list = $hlp->best();
+        $list = $hlp->best($data[1]['lang'],$data[1]['month'], 20);
 
-        $renderer->listu_open();
+        $renderer->listo_open();
+        $num_items=0;
         foreach($list as $item) {
+            if (auth_quickaclcheck($item['page']) < AUTH_READ) continue;
```

Comment:
Problem with that is that it requires disabling page caching. Thats why I wanted to remove every non-public link. Eg. do an explicit ACL check with no user.